### PR TITLE
Migrated Ical.Net.Tests project from NUnit3 to NUnit 4

### LINKED ISSUE #611: Migrate unit tests from NUnit v3 to v4
Migrating from `NUnit `v3 to v4 requires to convert Classic Assert to Constraint Model
@minichma  Expecting smooth migration following the [migration guide](https://docs.nunit.org/articles/nunit/release-notes/Nunit4.0-MigrationGuide.html)

### PR DESCRIPTION
**Pull request by axunonb**
* Migrate NUnit 3.14.0 to 4.2.2
* Add NUnit.Analyzers 4.3.0
* Convert Classic Assert to Constraint Model
* Introduce Assert.Multiple to group assertions, ensuring all are evaluated even if some fail.
* Simplify test case source return types to IEnumerable without a type argument
* Remove unused using directives and added necessary ones.

Test logic is left unchanged except for 7 tests in Exception context. 
Here all try...catch blocks are replaced with `Throws.` assertions.

Closes #611

### CODE REVIEW

**Comment by axunonb** [comment]
@minichma Please take a look, if time allows. Can we merge this PR?

**Comment by minichma** [comment]
Sure, will have a look right now.

File: Ical.Net.Tests/RecurrenceTests.cs [hunk 1]
```diff
@@ -38,14 +37,14 @@ int eventIndex
                 .OrderBy(o => o.Period.StartTime)
                 .ToList();
 
-            Assert.AreEqual(
-                dateTimes.Length,
-                occurrences.Count,
+            Assert.That(
+                occurrences,
+Has.Count.EqualTo(dateTimes.Length),
```
Comment (minichma) [inline Ical.Net.Tests/RecurrenceTests.cs]: indentation

File: Ical.Net.Tests/SimpleDeserializationTests.cs [hunk 2]
```diff
@@ -204,7 +231,9 @@ public void Encoding2()
             ProgramTest.TestCal(iCal);
             var evt = iCal.Events.First();
 
-            Assert.AreEqual(
+            Assert.That(
+evt.Attachments[0].ToString(),
```
Comment (minichma) [inline Ical.Net.Tests/SimpleDeserializationTests.cs:235]: indentation

File: Ical.Net.Tests/TodoTest.cs [hunk 3]
```diff
@@ -211,9 +211,9 @@ public void Todo7_1()
                 new CalDateTime(2006, 7, 1, 9, 0, 0),
                 new CalDateTime(2007, 7, 1, 9, 0, 0)).OrderBy(o => o.Period.StartTime).ToList();
 
-            Assert.AreEqual(
-                items.Count,
-                occurrences.Count);
+            Assert.That(
+                occurrences,
+Has.Count.EqualTo(items.Count));
```
Comment (minichma) [inline Ical.Net.Tests/TodoTest.cs:216]: indentation

**Review by minichma — DISMISSED** [review]
LGTM. Was just a very quick review, more an optical one. Found just a few formatting issues.

File: Ical.Net.Tests/TodoTest.cs (hunk 3 above)
Comment (axunonb) [reply Ical.Net.Tests/TodoTest.cs:216]: tnx

**Pull request merged**